staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -290,6 +301,15 @@ func (r *crdHandler) updateCustomResourceDefinition(oldObj, newObj interface{})
 	r.customStorageLock.Lock()
 	defer r.customStorageLock.Unlock()
 
+	// Add CRD to the establishing controller queue.
+	// For HA clusters, we want to prevent race conditions when changing status to Established,
+	// so we want to be sure that CRD is Installing at least for 5 seconds before Establishing it.
+	if apiextensions.IsCRDConditionTrue(newCRD, apiextensions.Installing) && r.masterCount > 1 {
```

Comment:
the `apiextensions.IsCRDConditionTrue(newCRD, apiextensions.Installing)` must guard both branch of the if-clause, mustn't it?